docs/resources/query.md

```diff
@@ -0,0 +1,162 @@
+---
+subcategory: "Databricks SQL"
+---
+# databricks_query Resource
+
+This resource allows you to manage [Databricks SQL Queries](https://docs.databricks.com/en/sql/user/queries/index.html).  It supersedes [databricks_sql_query](sql_query.md) resource - see migration guide below for more details.
+
+## Example Usage
+
+```hcl
+resource "databricks_directory" "shared_dir" {
+  path = "/Shared/Queries"
+}
+
+# This will be replaced with new databricks_query resource
+resource "databricks_query" "this" {
+  warehouse_id = databricks_sql_endpoint.example.id
+  display_name = "My Query Name"
+  query_text   = "SELECT 42 as value"
+  parent_path  = databricks_directory.shared_dir.path
+}
+```
+
+## Argument Reference
+
+The following arguments are available:
+
+* `query_text` - (Required, String) Text of SQL query.
+* `display_name` - (Required, String) Name of the query.
+* `warehouse_id` - (Required, String) ID of a SQL warehouse which will be used to execute this query.
+* `parent_path` - (Optional, String) The path to a workspace folder containing the query. The default is the user's home folder.  If changed, the query will be recreated.
+* `owner_user_name` - (Optional, String) Query owner's username.
+* `apply_auto_limit` - (Optional, Boolean) Whether to apply a 1000 row limit to the query result.
+* `catalog` - (Optional, String) Name of the catalog where this query will be executed.
+* `schema` - (Optional, String) Name of the schema where this query will be executed.
+* `description` - (Optional, String) General description that conveys additional information about this query such as usage notes.
+* `run_as_mode` - (Optional, String) Sets the "Run as" role for the object.
+* `tags` - (Optional, List of strings) Tags that will be added to the query.
+* `parameter` - (Optional, Block) Query parameter definition.  Consists of following attributes (one of `*_value` is required):
```

Comment:
Just curious: in other docs, these blocks are often documented as their own headings. Is there any need for consistency around these aspects?

Reply:
that's a good question - sometimes, moving to a separate block will add an unnecessary context switch. But agree, we need to unify this